WhisperAPI/WhisperAPI/Models/ConversationContext.cs

```diff
@@ -10,15 +11,30 @@ public ConversationContext(Guid chatkey, DateTime datetime)
         {
             this.ChatKey = chatkey;
             this.StartDate = datetime;
+            this.MessagesSuggestions = new List<MessageSuggestion>();
         }
 
         private ConversationContext()
         {
+            this.MessagesSuggestions = new List<MessageSuggestion>();
         }
 
         [Key]
         public Guid ChatKey { get; set; }
 
         public DateTime StartDate { get; set; }
+
+        public ICollection<MessageSuggestion> MessagesSuggestions { get; set; }
+
+        public string GetAllMessages()
+        {
+            var allMessages = string.Empty;
+            if (this.MessagesSuggestions != null)
+            {
+                allMessages = string.Join(" ", this.MessagesSuggestions.Select(m => m.Message));
```

Comment:
You check that the list of messages is not null, but it cannot be null because you initialize it in the constructor. But we never know, you should do a null check for the MessageSuggestion itself:

allMessages = string.Join(" ", this.MessagesSuggestions.Select(m => m?.Message)); 

Pay attention to the question mark I added in front of message:
https://stackoverflow.com/questions/28352072/what-does-question-mark-and-dot-operator-mean-in-c-sharp-6-0